  <launch>
  <!-- Arguments -->
  <arg name="camera_topic" default="/camera/rgb/image_color"/>
  <arg name="points_topic" default="/camera/depth_registered/points"/>
  <arg name="action_gui" default="false"/>
  <arg name="kinect_interactive_tf" default="false" />
  <arg name="freenect_driver" default="false" />

  <!--
  Important: in the kinect model.sdf in ~/.gazebo/models it has to be specified the kinect's plugin
  -->

  <!--
  <node pkg="tf" type="static_transform_publisher" name="upwards_camera_rgb_tf"
    args="0 0 0 0.70711 0.70711 0 0 camera_rgb_optical_frame upwards_camera_rgb_tf 100"> 
  </node>
  -->

  <node pkg="tf" type="static_transform_publisher" name="camera_link_tf" unless="$(arg kinect_interactive_tf)"
    args="0.9014760 -0.0594 0.7251720 -0.641407 0.032404 0.772933 0.011045 /iri_wam_link_base camera_link 100"> 
    <!--args="0.9014760 -0.0594 0.7251720 -0.6411 0.0380 0.7730 0.0043 /$(arg ROBOT)_link_base camera_link 20" -->
  </node>

  <include file="$(find freenect_launch)/launch/freenect.launch" if="$(arg freenect_driver)">
    <arg name="depth_registration" default="true" />
  </include>

  <node pkg="tf" type="static_transform_publisher" name="camera_rgb_optical_frame_tf" unless="$(arg freenect_driver)"
    args="0 0 0 -0.5 0.5 -0.5 0.5 camera_link camera_rgb_optical_frame 100"> 
  </node>

</launch>
